rust/src/lockfile.rs

```diff
@@ -104,16 +111,49 @@ struct LockfileRepoMetadata {
 }
 
 #[derive(Serialize, Deserialize, Debug)]
-struct LockedPackage {
-    evra: String,
-    #[serde(skip_serializing_if = "Option::is_none")]
-    digest: Option<String>,
+#[serde(untagged)]
+enum LockedPackage {
+    Evr {
+        evr: String,
+        digest: Option<String>,
+    },
+    Evra {
+        evra: String,
+        digest: Option<String>,
+    },
 }
 
 impl LockfileConfig {
     fn merge(&mut self, other: LockfileConfig) {
         self.packages.extend(other.packages);
     }
+
+    pub(crate) fn get_locked_packages(&self) -> CxxResult<Vec<crate::ffi::LockedPackage>> {
+        self.packages
+            .iter()
+            .map(|(k, v)| match v {
+                LockedPackage::Evr { evr, digest } => Ok(crate::ffi::LockedPackage {
+                    name: k.clone(),
```

Comment:
We can actually returned borrowed pointers BTW.  We haven't done this much yet, and nothing prevents the C++ side from violating the borrowing rules - *but* in cases where we're passing a lot of data I think it makes sense.  Don't need to do this now though.